nntrainer/models/model_loader.cpp

```diff
@@ -177,43 +177,66 @@ int ModelLoader::loadLayerConfigIniCommon(dictionary *ini,
                                           const std::string &layer_name,
                                           const std::string &layer_type) {
   int status = ML_ERROR_NONE;
+  int num_entries = iniparser_getsecnkeys(ini, layer_name.c_str());
 
-  try {
-    layer = nntrainer::createLayer(layer_type);
-  } catch (const std::exception &e) {
-    ml_loge("%s %s", typeid(e).name(), e.what());
-    status = ML_ERROR_INVALID_PARAMETER;
-  } catch (...) {
-    ml_loge("unknown error type thrown");
-    status = ML_ERROR_INVALID_PARAMETER;
+  ml_logd("number of entries for %s: %d", layer_name.c_str(), num_entries);
+
+  if (num_entries < 1) {
+    std::stringstream ss;
+    ss << "there are no entries in the layer section: " << layer_name;
+    throw std::invalid_argument(ss.str());
   }
-  NN_RETURN_STATUS();
 
-  unsigned int property_end =
-    static_cast<unsigned int>(Layer::PropertyType::unknown);
+  std::unique_ptr<const char *[]> key_refs(new const char *[num_entries]);
 
-  for (unsigned int i = 0; i < property_end; ++i) {
-    std::string prop = propToStr(i);
-    std::string value =
-      iniparser_getstring(ini, (layer_name + ":" + prop).c_str(), unknown);
+  if (iniparser_getseckeys(ini, layer_name.c_str(), key_refs.get()) ==
+      nullptr) {
+    std::stringstream ss;
+    ss << "failed to fetch key for layer: " << layer_name;
+    throw std::invalid_argument(ss.str());
+  }
 
-    /**! @todo: add following negative tc after #319
-     * 1. layer has empty prop -> throw std::invalid_argument
-     * 2. layer has not allowed property -> throw exception::not_supported
-     * 3. property value parse error -> throw std::invalid_argument
-     */
-    if (!strncmp(value.c_str(), unknown, strlen(unknown))) {
+  std::vector<std::string> properties;
```

Comment:
`properties.reserve(num_entries)`?

Reply:
actually `num_entries - 1` as `type` or `backbone` are always ommited.

Reply:
Changed the code to reserved at resource aquisition. Thks :)